azurerm/resource_arm_netapp_volume.go

```diff
@@ -0,0 +1,378 @@
+			return tf.ImportAsExistsError("azurerm_netapp_volume", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	creationToken := d.Get("creation_token").(string)
+	serviceLevel := d.Get("service_level").(string)
+	subnetId := d.Get("subnet_id").(string)
+	usageThreshold := int64(d.Get("usage_threshold").(int) * 1073741824)
+	exportPolicyRule := d.Get("export_policy_rule").(*schema.Set).List()
+
+	parameters := netapp.Volume{
+		Location: utils.String(location),
+		VolumeProperties: &netapp.VolumeProperties{
+			CreationToken:  utils.String(creationToken),
+			ServiceLevel:   netapp.ServiceLevel(serviceLevel),
+			SubnetID:       utils.String(subnetId),
+			UsageThreshold: utils.Int64(usageThreshold),
+			ExportPolicy:   expandArmNetAppVolumeExportPolicyRule(exportPolicyRule),
+		},
+	}
+
+	future, err := client.CreateOrUpdate(ctx, parameters, resourceGroup, accountName, poolName, name)
+	if err != nil {
+		return fmt.Errorf("Error creating NetApp Volume %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of NetApp Volume %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, accountName, poolName, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving NetApp Volume %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read NetApp Volume %q (Resource Group %q) ID", name, resourceGroup)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmNetAppVolumeRead(d, meta)
+}
+
+func resourceArmNetAppVolumeRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Netapp.VolumeClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	accountName := id.Path["netAppAccounts"]
+	poolName := id.Path["capacityPools"]
+	name := id.Path["volumes"]
+
+	resp, err := client.Get(ctx, resourceGroup, accountName, poolName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] NetApp Volumes %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading NetApp Volumes %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("account_name", accountName)
+	d.Set("pool_name", poolName)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if props := resp.VolumeProperties; props != nil {
+		d.Set("creation_token", props.CreationToken)
+		d.Set("service_level", props.ServiceLevel)
+		d.Set("subnet_id", props.SubnetID)
+
+		if props.UsageThreshold != nil {
+			d.Set("usage_threshold", *props.UsageThreshold/1073741824)
```

Comment:
what units is `usage_threshold` in? could we make the name `usage_threshold_in_units`

Reply:
I am confused. You want this field name is updated to "usage_threshold_in_units" or "storage_quota_in_gb"? I think "storage_quota_in_gb" is preferred so that I updated this field name to "storage_quota_in_gb".